doc/ui-guidelines.md

```diff
@@ -252,3 +265,150 @@ GOOD:
 ```
 
 So, now the screen component function will never be invoked, and `component` and `component2` will be re-rendered only when `label` or `label2` have changed.
+
+## Notes on Memoization
+
+Memoization is an optimization technique used to speed up computer programs by storing the results of expensive function calls and returning the cached result when the same inputs occur again. It is a specific form of caching where results of function calls are stored based on their input arguments. However, it's important to note that this technique introduces memory overhead, as it requires additional space to store the results of function calls. While it is a powerful tool for improving performance, we must balance its benefits against the increased memory usage, especially in resource-constrained environments.
+
+Memoization in React is used to ensure that a component or a computation does not re-render or re-calculate `unnecessarily` when its input props or dependencies have not changed. This technique can significantly improve the performance of React applications, especially for `expensive`, `computation-heavy` operations or components that `render frequently`.
+
+In React, we generally have three util functions for memoizing :
+- React.memo (or `rn/memo` in the case of our codebase):
+Memoizes a component's render output based on its props. It can optionally take a comparison function to customize how changes in props are detected.
+ e.g:
+  ```clojure
+  (def pure-component
+    (rn/memo (fn [props]
+              [view props])
+            (fn [prev-props next-props]
+              (= (:value prev-props) (:value next-props)))))
+  ```
+
+- React.useMemo (or `rn/use-memo` in the case of our codebase):  Memoizes a computed value so that it does not need to be re-calculated on every render, given that its dependencies haven't changed. e.g:
+  ```clojure
+  (defn component [props]
```

Comment:
Minor neatpick, 
this example can be a bit misleading. In re-frame components won't re-render unless their `props`, so if you pass `props` to the memoization function, it will always re-run since the caching layer is identical (it would only be used if props flip back to a previous already rendered state).

maybe a more usual case is something like:

```
(defn component [{:keys [something] :as props}]
   (let [memoized-value (rn/use-memo ..... [something])]
    ....
```
That's most of the times what we want i.e the component will re-render because certain props have changed, but we memoized based on a subset of the props.